### DISCUSSION TITLE
XYZ tristimulus bitmap

### DISCUSSION THREAD

**Discussion by ssh4net** [mitsuba-renderer/mitsuba3#1259] [Q&A]
What white point is used for XYZ tristimulus values?
D50 or D65?

I need to import wider than sRGB gamut texture to Mitsuba and found that it support XYZ color space. But no clue what white point used. Color science usually use D50 but sRGB is D65.

Thank you.

**Comment by merlinND** [comment]
Hello @ssh4net,

I believe we are using D65, but maybe @wjakob can confirm.

**Comment by ssh4net** [comment]
@merlinND sorry, how should i define that this bitmap is XYZ values?

```
    'shape': {
        'type': 'rectangle',
        'to_world': mi.ScalarTransform4f.scale([1.0, 2.2*0.21, 2.2*0.29]) @
        mi.ScalarTransform4f(
            [[ 0, 0, 1, 0],
             [ 0, 1, 0, 0],
             [-1, 0, 0, 0],
             [ 0, 0, 0, 1]]
        ),
        'bsdf': {
            'type': 'diffuse',
            'reflectance': {
                'type': 'bitmap',
                'filename': r'x:\FTP\Vlad\SPECTRAL\ColorChartSG_XYZ_D65.exr'
            },
        }
    }
```

This one load image as RGB as I understand

**Reply by merlinND** [reply]
If the channels of your EXR image are called x, y and z the Bitmap's pixel format will be automatically detected as XYZ:

https://github.com/mitsuba-renderer/mitsuba3/blob/d7fdd68feb4fa0e481857f0d860f2ab796217f44/src/core/bitmap.cpp#L1077-L1081

Otherwise, it looks to me like you'd have to build a new `mi.Bitmap` from the image you load, something like:
```py
bmp_loaded = mi.Bitmap("my_file.exr")
bmp_xyz = mi.Bitmap(np.array(bmp_loaded), pixel_format=mi.PixelFormat.XYZ)
```
Then, you can pass `bmp` into your scene dict:

```py
           'reflectance': {
                'type': 'bitmap',
                'bitmap': bmp_xyz,
            },
```

Not super convenient. I think it would be nicer to be able to:
1. Override `pixel_format` in the `Bitmap(Stream *stream, ...)` and `Bitmap(const fs::path &path, ...)` constructors.
2. Provide a `pixel_format` for override to the `Bitmap` texture props.

What do you think @njroussel ?

**Reply by njroussel** [reply]
Hi :) 

Option 1 is nicer, I think. And your suggestion is the only current workaround I know of.

**Reply by ssh4net** [reply]
@merlinND

Using this way
```
bmp_loaded = mi.Bitmap("my_file.exr")
bmp_xyz = mi.Bitmap(np.array(bmp_loaded), pixel_format=mi.PixelFormat.XYZ)
```

Code give a warning:
2024-08-14 11:53:29 WARN wrk8 [BitmapTexture] BitmapTexture: texture named "" contains pixels that exceed the [0, 1] range!

Why can this warning be? The XYZ data in file is in 0~1 range. And the 
`np.array(bmp_xyz ).tofile('d:/text.txt,sep=' ')` do not have any values higher than 1.0 or less than 0.0
